app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue

```diff
@@ -41,15 +44,24 @@ const dynamicTags = computed<Tag[]>(() => {
   const config = props.config
   if (config == null) return []
   const [_, widgetConfig] = config.find(([name]) => name === props.input.info?.name) ?? []
-  if (widgetConfig && widgetConfig.kind == 'Single_Choice') {
-    return widgetConfig.values.map((value) => ({
-      kind: 'Dynamic',
-      label: value.label || value.value,
-      value: value.value,
-    }))
+  if (widgetConfig == null) return []
+  let values: Choice[]
+  if (
+    props.input instanceof VectorItem &&
+    widgetConfig.kind === 'Vector_Editor' &&
+    widgetConfig.item_editor.kind === 'Single_Choice'
+  ) {
+    values = widgetConfig.item_editor.values
+  } else if (widgetConfig.kind === 'Single_Choice') {
+    values = widgetConfig.values
```

Comment:
It's very patchy. Now every widget must consider if it happens to be inside vector editor. I think the ListEditor should just pass `item_editor` as dynamicConfig of its elements. Similarly, it's the WidgetFunction/Application which should extract the piece of config for every argument.